components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1694,7 +1891,7 @@ bool KeyringService::UpdateNameForHardwareAccountSync(
     const std::string& address,
     const std::string& name,
     mojom::CoinType coin) {
-  auto keyring_id = GetKeyringIdForCoin(coin);
+  auto keyring_id = GetMainKeyringIdForCoin(coin);
```

Comment:
Shouldn't we use `GetHardwareKeyringId` here? And this also implied that our test coverage for this is insufficient so we didn't catch it.